### PR TITLE
French Meal Vouchers

### PR DESCRIPTION
**Pull request by araratthehero**
## Description
Implementation of French meal vouchers.

The implementation extends the Giftcard logic, since French meal vouchers are handled like giftcards, with an additional field. To keep them separate, a new module has been created.

## Pending
We are still waiting for test cards to finalize the testing of the integration.

## Checklist 
- [x] PR is labelled 
- [x] Code is unit tested
- [x] Changes are tested manually

COAND-683

### CODE REVIEW

File: ui-core/src/main/java/com/adyen/checkout/ui/core/internal/ui/view/ExpiryDateInput.kt
Comment (jreij) [inline ui-core/src/main/java/com/adyen/checkout/ui/core/internal/ui/view/ExpiryDateInput.kt:1]: Can we make this restricted?

File: ui-core/src/main/java/com/adyen/checkout/ui/core/internal/ui/view/SecurityCodeInput.kt
Comment (jreij) [inline ui-core/src/main/java/com/adyen/checkout/ui/core/internal/ui/view/SecurityCodeInput.kt:1]: Can we make this restricted?

File: meal-voucher-fr/src/main/java/com/adyen/checkout/mealvoucherfr/MealVoucherFRConfiguration.kt
Comment (jreij) [inline meal-voucher-fr/src/main/java/com/adyen/checkout/mealvoucherfr/MealVoucherFRConfiguration.kt:1]: Let's add a method for this configuration inside `DropInConfiguratiion.Builder`

File: ui-core/src/main/java/com/adyen/checkout/ui/core/internal/ui/view/ExpiryDateInput.kt
Comment (araratthehero) [reply ui-core/src/main/java/com/adyen/checkout/ui/core/internal/ui/view/ExpiryDateInput.kt:1]: If we restrict the entire class, then the class can not be resolved in xml files, without suppressing the `AndroidUnresolvableTag` in xml files.
I will restrict the constructors

File: ui-core/src/main/java/com/adyen/checkout/ui/core/internal/ui/view/SecurityCodeInput.kt
Comment (araratthehero) [reply ui-core/src/main/java/com/adyen/checkout/ui/core/internal/ui/view/SecurityCodeInput.kt:1]: Same for this one

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -11,6 +11,10 @@
 ## New
 - Added support for 6 more locales: Catalan (ca-ES), Icelandic (is-IS), Bulgarian (bg-BG),
   Estonian (et-EE), Latvian (lv-LV) and Lithuanian (lt-lT).
+- French meal vouchers are now available with the following payment method types:
```
Comment (jreij) [inline RELEASE_NOTES.md:14]: Can we link to the documentation just like with twint?

Comment (ozgur00) [reply RELEASE_NOTES.md]: Done.

**Pull request merged**